bamboo/unit_tests/test_unit_datastore_imagenet.py

```diff
@@ -152,64 +184,44 @@ def func(cluster, exes, dirname, weekly):
-        # of metric values.
-        assert len(baseline_metrics) == len(datastore_metrics), \
-            'baseline and data store experiments did not run for same number of epochs'
-        for i in range(len(datastore_metrics)):
-            x = baseline_metrics[i]
-            xhat = datastore_metrics[i]
-            eps = np.finfo(np.float32).eps
-            ceillogx = int(math.ceil(math.log10(x)))
-            assert abs(x-xhat) < max(8*eps*x, 1.5*10**(ceillogx-6)), \
-                'found large discrepancy in metrics for baseline and data store experiments'
-
-        # Check if the output 'data_store_profile_train.txt' file
-        # contains the entries specified in the 'profile_tests' input param
-        #
-        # TODO:
+        # Run LBANN experiments with data store
+        for i in range(len(datastore_test_funcs)) :
+            r = create_datastore_test_func(datastore_test_funcs[i], baseline_metrics, cluster, exes, dirname)
+            assert len(r) == 0, r
+
+        #assert all_tests_passed, '\n' + ' '.join(results)
 
     # Return test function from factory function
     func.__name__ = test_name
     return func
-
-
+ 
 # Create test functions that can interact with PyTest
-def make_test(name, args=[]) :
-  return (tools.create_tests(
+def make_test(name, test_by_platform_list=[], args=[]) :
+    test_list = tools.create_tests(
             setup_experiment,
             __file__,
             nodes=num_nodes,
             test_name_base='test_unit_datastore_imagenet_' + name,
-            lbann_args=args))
+            lbann_args=args)
+    if test_by_platform_list != [] :
+        for i in range(len(test_list)) :
+          test_by_platform_list[i].append(test_list[i])
+    else :
+      return test_list
 
 baseline_tests = make_test('nodatastore')
-
-datastore_tests = []
+datastore_tests = [[] for j in range(len(baseline_tests))]
 
 #local cache with explicit loading
-datastore_tests.append(make_test('data_store_cache_explicit', ['--data_store_cache', '--data_store_profile']))
+ds = make_test('data_store_cache_explicit', datastore_tests, ['--data_store_cache', '--data_store_profile'])
 
 #local cache with preloading
-datastore_tests.append(make_test('data_store_cache_preloading', ['--data_store_cache', '--preload_data_store', '--data_store_profile']))
-
-#explicit loading
-datastore_tests.append(make_test('use_data_store', ['--use_data_store', '--data_store_profile']))
+ds = make_test('data_store_cache_preloading', datastore_tests, ['--data_store_cache', '--preload_data_store', '--data_store_profile'])
 
-#preloading
-datastore_tests.append(make_test('preload_data_store', ['--preload_data_store', '--data_store_profile']))
+#local cache with preloading -- this will fail; only for use
+#during test development
+##ds = make_test('data_store_cache_preloading', datastore_tests, ['--data_store_cache', '--preload_data_store', '--data_store_profile', '--data_store_fail'])
```

Comment:
We should remove clutter. That said, I do like the idea of adding tests to make sure failures happen when we give them bogus inputs.